playwright/tests/e2e/steps/transactions-overview.steps.ts

```diff
@@ -0,0 +1,221 @@
+import { createBdd } from "playwright-bdd";
+import { expect } from "@playwright/test";
+
+import { koiosApi } from "playwright/api/call-koios/koios.api";
+
+import { transactionOverviewPage } from "../../../pages/transaction-overview.page";
+import { transactionDetailPage } from "../../../pages/transaction-detail.page";
+
+const { Given, When, Then } = createBdd();
+
+let addressId: string | null;
+let transactionHash: string | null;
+
+Given(/^the user is in the general dashboard page in explorer portal test$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToDashboard();
+});
+When(/^the user selects the Transactions option inside the Blockchain drop down menu$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToEpochsFromSidebar();
+});
+Then(
+  /^the user should see the transactions page containing the Search bar and Transactions table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).searchBarOnTransaction();
+    await transactionOverviewPage(page).checkTransactionsTable();
+  }
+);
+
+Given(/^the user is in the transactions page in explorer portal site$/, async ({ page }) => {
+  await transactionOverviewPage(page).goToTransactions();
+});
+When(
+  /^the user selects any data of the row execpt the transaction hash, block number, epoch number or addresses hash of one of the transactions record in the table$/,
+  async ({ page }) => {
+    await transactionOverviewPage(page).openWidget();
+  }
+);
+Then(
+  /^the user should see the Info widget data with the respective data of the selected transaction$/,
+  async ({ page, request }) => {
+    const trxHash = await page.getByTestId("transaction.table.value.txhash#0");
```

Comment:
Steps methods should not have interactions over the pages locators, I understand here you are getting the tx hash from the opened widget and then use that Tx hash for the Koios request. 

A better way to handle this is in the previous steps the `when` step where you open the widget of a transaction, in that moment you capture and return the Tx hash that you will use in the `Then` step. 

Keep in mind that interact with locators is just an action of the Pages classes and the Steps just call functions that has the logic with this locators